### PR TITLE
Implement `output_size` for `osquery_schedule`

### PR DESCRIPTION
**Pull request by sharvilshah**
Addresses #7147

This PR implements a simple `output_size` column for scheduled queries, while being aware of the newer typed `RowData`.

Continuation of #7377

### CODE REVIEW

File: osquery/sql/sqlite_util.cpp [hunk 1]
```diff
@@ -223,6 +223,32 @@ class StringEscaperVisitor : public boost::static_visitor<> {
...
+    return size;
+  }
+
+  SizeVisitor() {
+    size = 0;
+  }
+
+ private:
+  uint64_t size;
+};
```
Comment (Smjert) [inline osquery/sql/sqlite_util.cpp:250]: nit: prefer
```suggestion
private:
  uint64_t size{0};
};
```
or `SizeVisitor() : size(0)`

**Review by Smjert — APPROVED** [review]
Thank you for moving forward with this!
Only a minor nit ;)

File: osquery/core/sql/query_performance.h [hunk 2]
```diff
@@ -34,6 +34,9 @@ struct QueryPerformance {
 
   /// Average memory differentials. This should be near 0.
   unsigned long long int average_memory{0};
+
+  /// Total bytes for the query
+  unsigned long long int output_size{0};
```
Comment (directionless) [inline osquery/core/sql/query_performance.h:39]: The comment on `average_memory` implies it's the memory used by the differentials, though I always thought it was the memory used by osquery?

File: specs/utility/osquery_schedule.table [hunk 3]
```diff
@@ -11,7 +11,7 @@ schema([
     Column("denylisted", INTEGER, "1 if the query is denylisted else 0",
         aliases=["blacklisted"]), # 'blacklist' now deprecated
     Column("output_size", BIGINT,
-      "Total number of bytes generated by the query"),
+      "Cumulative total number of bytes generated by the resultant rows of the query"),
```
Comment (directionless) [inline specs/utility/osquery_schedule.table:14]: Is this exact or approximate?

**Review by directionless — COMMENTED** [review]
I'm excited to see this

File: osquery/core/sql/query_performance.h (hunk 2 above)
Comment (Smjert) [reply osquery/core/sql/query_performance.h:39]: Shameless plug: I have changed that comment and the column description here https://github.com/osquery/osquery/pull/7438

Still have to open a Blueprint but it seems to be that that counter is not so useful; what the user is most likely interested in is the peak memory usage, which has to be gathered by continuosly sampling memory usage during the query execution.

File: specs/utility/osquery_schedule.table (hunk 3 above)
Comment (sharvilshah) [reply specs/utility/osquery_schedule.table:14]: Exact bytes!

**Comment by sharvilshah** [comment]
I think this is good to go and requires a final review. Can we also add this to the next milestone?

**Review by Smjert — APPROVED** [review]

**Pull request merged**